source/tests/system/visa_driver_api_tests.cpp

```diff
@@ -153,6 +153,149 @@ TEST_F(VisaDriverApiTest, SetNonTcpipAttribute_ReturnsAttributeError)
   );
 }
 
+class VisaDriverLoopbackTest : public VisaDriverApiTest {
+  public:
+  VisaDriverLoopbackTest()
+  : portNumber_("1234")
+  , instrument_descriptor_("TCPIP0::localhost::" + portNumber_ + "::SOCKET")
+  {
+  }
+  virtual ~VisaDriverLoopbackTest()
+  {
+  }
+
+  void SetUp() override
+  {
+    start_server_session(portNumber_);
+    initialize_driver_session(instrument_descriptor_);
+  }
+
+  void TearDown() override
+  {
+    close_driver_session();
+    close_server_session();
+  }
+
+  void start_server_session(const std::string& portNumber)
+  {
+  #ifdef _WIN32
+    WSADATA wsa_data;
+    WSAStartup(MAKEWORD(2, 2), &wsa_data);
+  #endif
+    server_fd_ = socket(AF_INET, SOCK_STREAM, 0);
+    EXPECT_NE(-1, server_fd_);
+    struct sockaddr_in server_address;
+    server_address.sin_family = AF_INET;
+    server_address.sin_addr.s_addr = inet_addr("127.0.0.1");
+    server_address.sin_port = htons(std::stoi(portNumber));
```

Comment:
You should specify port 0 when starting the server. Then use `getsockname` for example to find the port that the OS assigned you. It will be different on every run, but that's OK.